elrond-wasm/src/types/managed/basic/big_int_cmp.rs

```diff
@@ -29,31 +29,33 @@ impl<M: ManagedTypeApi> Ord for BigInt<M> {
     }
 }
 
-fn cmp_i64<M: ManagedTypeApi>(bi: &BigInt<M>, other: i64) -> Ordering {
+fn cmp<M: ManagedTypeApi, T>(bi: &BigInt<M>, other: T) -> Ordering
+where
+    T: TryInto<i64> + PartialEq<T> + From<u8>,
+{
     let api = M::managed_type_impl();
-    if other == 0 {
+    if other == 0u8.into() {
```

Comment:
Isn't it more natural to convert to i64 the first thing and then use the i64 from there on?

I would even split this function in 2:
fn cmp_i64(..., other: i64)
fn cmp_conv_i64<T: TryInto<i64>>(..., other: T) 

I know there's a bit of refactoring there, but it's not that complicated

Reply:
Actually I think we should be unifying this logic with BigUint, it's pretty much the same code

Reply:
Done. Implemented as suggested.